### PR TITLE
[Fix Bug] Voice Ava (Premium) not found, add dynamic module for macos tts

### PR DESCRIPTION
**Pull request by gabkk**
### Background
This pull request aims to improve the macOS TTS voice support in the Auto-GPT repository. The previous implementation had a fixed voice for the second agent, "Ava (Premium)", which might not be available on all systems. This PR provides a more dynamic approach to handling macOS TTS voices.

### Changes
1. The MacOSTTS class has been updated to include a `_setup()` method that fetches the available voices on the system and sets up the default and custom voices accordingly.
2. Added a `get_voice_names()` method to the MacOSTTS class to retrieve available voice names on the system.
3. Updated the `_speech()` method in the MacOSTTS class to use voices from the `_voices` list instead of hardcoding voice names.
4. Updated the `.env.template` and `config.py` files to include new environment variables for specifying custom macOS TTS voices.

### Documentation
In-code comments and descriptions have been added to the MacOSTTS class methods, explaining their purpose, functionality, input arguments, and return values.

### Test Plan
1. Set up the Auto-GPT repository on a macOS system.
2. Configure the `.env` file to enable macOS TTS support and specify custom voices using the provided environment variables.
3. Run the Auto-GPT application and test with various prompts to ensure that the macOS TTS voices are working correctly and switching between different voices as expected.

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes.

### CODE REVIEW

**Comment by gabkk** [comment]
@Torantulino 

If you have time to check this PR, it doesn't have any breaking change and it will fix voice issues on the second ai agent on Macos  🙏

**Comment by Wladastic** [comment]
I think default for macos should be "say text" without any voice then.
I had added a file that determines the available voices as well, it got deleted at some point.
There are "bettter quality" options to use by gathering the xml file within siri in the system, that's where I got mine from:


catherine
martha
maeve
riya
nora
leona
gordon
arthur
aidan
akash
aaron
xander

to use them you have to add the path as prefix

**Comment by gabkk** [comment]
Hi @Wladastic 

Yeah you are right i can update my PR for the main voice to not use any specific voice.
But the problem currently is the second voice "Ava premium" is not recognised and when auto-GPT spin up an other agent it's just silent.

i am using `say -v '?'` which is listing the available voice option on your Os.
So you can provide any voice listed from the `say -v '?'`

It also better to have 2 voices when using the vocal option so you can easily know which agent is speaking.

I ll update my PR to remove the voice for the first agent and list all the available voice when using the macostts

File: .env.template [hunk 1]
```diff
@@ -165,8 +165,13 @@ OPENAI_API_KEY=your-openai-api-key
...
 ### MAC OS
-## USE_MAC_OS_TTS - Use Mac OS TTS or not (Default: False)
-# USE_MAC_OS_TTS=False
+# USE_MAC_OS_TTS - Use Mac OS TTS or not (Default: False)
+# Run this command in your terminal to see available voices: say -v '?'
+# MACOS_VOICE_1 voice used by the main assistant (Optional)
+# MACOS_VOICE_2 voice used by the second agent (Optional)
+USE_MAC_OS_TTS=False
+MACOS_VOICE_1=Victoria
+MACOS_VOICE_2=Alex
```
Comment (Wladastic) [inline .env.template]:
```suggestion
# USE_MAC_OS_TTS=False
# MACOS_VOICE_1=Victoria
# MACOS_VOICE_2=Alex
```

Comment (gabkk) [reply .env.template]: 1. Merged master into my branch
2. Commented out option in .env.example

@Wladastic

File: autogpt/speech/macos_tts.py [hunk 2]
```diff
@@ -1,21 +1,59 @@
...
     def _setup(self) -> None:
-        pass
+        """Set up the voices for MacOSTTS.
+
+        Returns:
+            None: None
+        """
+        cfg = Config()
+        available_voices = self.get_voice_names()
+        default_voices = ["Alex", "Victoria"]
```
Comment (Wladastic) [inline autogpt/speech/macos_tts.py:20]: As default first voice I would set it to use say without -v which will be the default of the system. 
i found that using that gives the best voice quality

Comment (gabkk) [reply autogpt/speech/macos_tts.py:20]: Those are the default voices for extra agents.

The default voice is not using `-v`

```
        if voice_index == 0:
            os.system(f'say "{text}"')
```

Comment (Wladastic) [reply autogpt/speech/macos_tts.py:20]: correct

Comment (Wladastic) [reply autogpt/speech/macos_tts.py:20]: AH seems I misread, nice!

**Review by Wladastic — DISMISSED** [review]

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Pull request closed without merging**